elks/arch/i86/kernel/irqtab.S

```diff
@@ -254,7 +274,9 @@ updct:
         pop     %bx
 
         pop     %ax             // Saved IRQ
+#ifndef CONFIG_ARCH_SWAN
         cli                     // Disable interrupts to avoid reentering ISR
+#endif
```

Comment:
This CLI may still want to stay as a result of the potential issues with the kernel possibly having enabled interrupts inadvertently (for Swan only). The CLI is very important here because otherwise there is the possibility of overflowing the kernel stack. I would suggest leaving this one in, just in case.